src/main/java/lotto/domain/WinningNumbers.java

```diff
@@ -0,0 +1,49 @@
+package lotto.domain;
+
+import java.util.List;
+
+public class WinningNumbers {
+
+    private final LottoTicket winningTicket;
+    private final LottoNumber bonusNumber;
+
+    public WinningNumbers(List<LottoNumber> winningNumbers, LottoNumber bonusNumber) {
+        this(new LottoTicket(winningNumbers), bonusNumber);
+    }
+
+    public WinningNumbers(LottoTicket winningTicket, LottoNumber bonusNumber) {
+        validateWinningNumber(winningTicket, bonusNumber);
+        this.winningTicket = winningTicket;
+        this.bonusNumber = bonusNumber;
+    }
+
+    public Prize checkWinning(LottoTicket lottoTicket) {
+        int matchCount = winningTicket.compare(lottoTicket);
+        if (matchCount <= 2) {
+            return Prize.NOTHING;
+        }
+        return getPrize(lottoTicket, matchCount);
+    }
+
+    private Prize getPrize(LottoTicket lottoTicket, int matchCount) {
+        if (matchCount == 3) {
+            return Prize.FIFTH;
+        }
+        if (matchCount == 4) {
+            return Prize.FOURTH;
+        }
+        if (matchCount == 5 && !lottoTicket.contains(bonusNumber)) {
+            return Prize.THIRD;
+        }
+        if (matchCount == 5 && lottoTicket.contains(bonusNumber)) {
+            return Prize.SECOND;
+        }
+        return Prize.FIRST;
+    }
```

Comment:
`일치하는 갯수`, `보너스볼 일치 여부` 등에 따라 적절한 Prize를 찾아서 반환해주는 역할이 WinningNumbers의 역할이라고 볼 수 있을까요? 🤔  물론 그렇다고 생각할 수도 있겠지만 그렇게되면 WinningNumbers가 Prize의 세부적인 구현사항에 의존하게됩니다. 

이 역할을 Prize 객체에게 위임해보면 어떨까요? 🙌
```suggestion
    public Prize checkWinning(LottoTicket lottoTicket) {
        int matchCount = winningTicket.compare(lottoTicket);
        boolean hasBonusBall = lottoTicket.contain(bonusBall);
        
        return Prize.matchOf(matchCount, hasBonusBall);
    }
```

Reply:
`TODO`로 적어주신 내용을 보니 이미 아래와 같은 고민을 하고계셨군요? 👍💯

> 응집성을 높이기 위해 WinningNumbers의 getPrize() 메소드를 Prize에 넣어 처리할까 생각중(가능한지는 확인해봐야 할 듯)